File: week6/psets/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")

    lettercount = letterCount(text)
    wordcount = wordCount(text)
    sentencecount = sentenceCount(text)

    # 0.0588 * LettersPer100Words - 0.296 * SentencesPer100Words - 15.8

    lpw = lettercount * 100 // wordcount
    spw = sentencecount * 100 // wordcount

    grade = round(0.0588 * lpw - 0.296 * spw - 15.8)

    if grade < 1:
        return "Before Grade 1"
    elif grade > 16:
        return "Grade 16+"

    return f"Grade {grade}"

def letterCount(txt):
    count = 0
    for i in range(len(txt)):
        if txt[i].isalpha():
            count += 1
    return count

def wordCount(txt):
    count = 0
    foundLetter = False
    for i in range(len(txt)):
        if txt[i] == ' ':
            foundLetter = False
        else:
            if not(foundLetter):
                count += 1
                foundLetter = True
    return count

def sentenceCount(txt):
    count = 0
    for i in range(len(txt)):
        if txt[i] == '?' or txt[i] == '!' or txt[i] == '.':
            count += 1
    return count

print(main())

File: psets/week2/readability.c
#include <math.h>
#include <string.h>
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>

int count_letters(string word);
int count_words(string word);
int count_sentences(string word);

int main(void)
{
    int sentenceCount;
    int letterCount;
    int wordCount;
    int lpw;
    int spw;

    string text = get_string("Text: ");

    letterCount = count_letters(text);
    wordCount = count_words(text);
    sentenceCount = count_sentences(text);

    lpw = letterCount  * 100 / wordCount;
    spw = sentenceCount * 100 / wordCount  ;

    // Formula is index = 0.0588 * lpw - 0.296 * spw - 15.8
    float calculation = 0.0588 * lpw - 0.296 * spw - 15.8;
    int index = round(calculation);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1){
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}









int count_letters(string word)
{
    int count = 0;
    for(int i = 0; i < strlen(word); i++)
    {
        if (isalpha(word[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string word)
{
    int count = 0;
    bool foundletter = false;
    for (int i = 0; i < strlen(word); i++)
    {
        if (word[i] == ' ')
        {
            foundletter = false;
        }
        else {
            if (foundletter == false)
            {
                count++;
                foundletter = true;
            }
        }
    }
    return count;
}

int count_sentences(string word)
{
    int count = 0;
    for (int i = 0; i < strlen(word); i++)
    {
        if (word[i] == '!' || word[i] == '?' || word[i] == '.')
        {
            count++;
        }
    }
    return count;
}
